### DISCUSSION TITLE
Home Assistant Map generation

### DISCUSSION THREAD

**Discussion by creatvty** [Hypfer/Valetudo#726] [Q&A / Support] [answered]
Hi,

so as the map is now generated by the lovelace-valetudo-map-card I have the issue, that I can not access the PNG itself anymore. I used to have an automation active that would send the generated map via telegram bot. If I am not mistaken, this is currently not possible because the lovelace-valetudo-map-card is not storing the map anywhere but generating it on the fly. Is there any way of accessing the map in the generated format like before?

Kind regards

**Comment by Hypfer** [comment, marked as answer]
Map to png rendering isn't part of Valetudo anymore since.. 0.6? I think

You need to run this companion service
https://github.com/Hypfer/Icantbelieveitsnotvaletudo

which will render the map and push it to another mqtt camera entity.

**Comment by creatvty** [comment]
Thanks for your quick reply. I was a little confused that the vacuum is providing the `camera.map` entity directly and lovelace-valetudo-map-card was rendering the map from this data. Whereas Icantbelieveitsnotvaletudo is providing the `camera.robot_rendered_map` entity. Also I had a small mistake in my config of Icantbelieveitsnotvaletudo. That's why it did not send the rendered map.
